mlbench_core/evaluation/pytorch/inference/beam_search.py

```diff
@@ -0,0 +1,293 @@
+import torch
+
+from mlbench_core.dataset.translation.pytorch.config import BOS, EOS
+
+
+class SequenceGenerator:
```

Comment:
so you implement it yourself or just borrow from another library? a bit worried about the efficiency of the beam search proposed here.

Reply:
This has been taken directly from NVIDIA code. It generally uses the `beam_search` instead of `greedy_search`.

Reply:
good to mark all existing code used clearly as such, then it also makes it easier in the future to upgrade and detangle things

Reply:
I added a comment with the link to the repo where it was taken from.